### PR TITLE
Storage model tests

### LINKED ISSUE #24: Add missing tests for the logic associated to the Storage model
- https://github.com/chris-zen/htsget-mvp/blob/main/htsget-search/src/storage/mod.rs#L136
- https://github.com/chris-zen/htsget-mvp/blob/main/htsget-search/src/storage/mod.rs#L161

### PR DESCRIPTION
**Pull request by mmalenic**
Add storage model tests, closes #24.

### CODE REVIEW

**Review by brainstorm — APPROVED** [review]
Fantastic, thanks Marko!

File: htsget-search/src/htsget/bam_search.rs [hunk 1]
```diff
@@ -3,8 +3,9 @@
 
 use std::{fs::File, path::Path};
 
-use bam::bai::index::reference_sequence::bin::Chunk;
 use noodles_bam::{self as bam, bai};
+use noodles_bgzf::index::Chunk;
```
Comment (brainstorm) [inline htsget-search/src/htsget/bam_search.rs:7]: Minor: I'd collapse those two in one line import.

File: htsget-search/src/htsget/bam_search.rs [hunk 2]
```diff
@@ -276,11 +277,11 @@ where
 
 #[cfg(test)]
 pub mod tests {
-
-  use super::*;
   use crate::htsget::Headers;
   use crate::storage::local::LocalStorage;
 
+  use super::*;
```
Comment (brainstorm) [inline htsget-search/src/htsget/bam_search.rs:283]: Why this move?

**Review by brainstorm — APPROVED** [review]

File: htsget-search/src/htsget/bam_search.rs (hunk 2 above)
Comment (mmalenic) [reply htsget-search/src/htsget/bam_search.rs:283]: This seems to have been an auto import optimize by my IDE, however it does match the style of the other tests.
I can change it back if you would like?

Comment (brainstorm) [reply htsget-search/src/htsget/bam_search.rs:282]: No, it's alright, I was just curious on whether it had to do with something else :)

**Pull request merged**